internal/dataplane/kongstate/kongstate.go

```diff
@@ -73,6 +73,18 @@ func (ks *KongState) FillConsumersAndCredentials(
 		c.K8sKongConsumer = *consumer
 		c.Tags = util.GenerateTagsForObject(consumer)
 
+		// Get consumer groups
+		for _, cgName := range consumer.ConsumerGroups {
+			cg, err := s.GetKongConsumerGroup(consumer.Namespace, cgName)
+			if err != nil {
+				failuresCollector.PushResourceFailure(fmt.Sprintf("non-existing consumer group: %q", err), consumer)
+				continue
+			}
+			c.ConsumerGroups = append(c.ConsumerGroups, kong.ConsumerGroup{
```

Comment:
Have we already included code to prevent the `consumer_group` to be sent to Kong gateways which does not support this field? I think only checking them in validating webhook is not enough, because it is possible that webhook is not installed.

Reply:
Not yet. You're right. I want to do it together with a webhook, to gracefully skip them and not break a config. It's tracked on this list https://github.com/Kong/kubernetes-ingress-controller/issues/4411 so I won't forget

Reply:
This does not populate anything in the config YAML if none are configured, correct? A lightweight documentation-only approach saying "this requires Kong Enterprise 3.4+" and a config apply error if you don't heed the docs is fine.

We use that for most everything that's not universally supported, e.g. that's what you'll hit if you try and apply an Enterprise-only KongPlugin to an OSS instance. Users generally don't attempt to use features that aren't available to them.